src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const initialValue: GroupsMap<T> = {};
+
+  const result: GroupsMap<T> = items.reduce((prev, value) => {
+    const groupKey = String(value[key]);
+    const copy = { ...prev };
+
+    if (!(groupKey in prev)) {
+      copy[groupKey] = [value];
+    } else {
+      copy[groupKey].push(value);
+    }
```

Comment:
it makes more sense to me
```suggestion
    if (groupKey in prev) {
      copy[groupKey].push(value);
    } else {
      copy[groupKey] = [value];
    }
```

Reply:
from permutation - the amount does not change.
I think it is not important to change.
let's leave it to the discretion of the mentor)